src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -22,213 +25,210 @@ public class StakeValidator
         /// <summary>Instance logger.</summary>
         private readonly ILogger logger;
 
-        /// <summary>Class logger.</summary>
-        private static readonly ILogger clogger;
-
-        /// <summary>Specification of the network the node runs on - regtest/testnet/mainnet.</summary>
-        private readonly Network network;
-
         /// <summary>Database of stake related data for the current blockchain.</summary>
         private readonly StakeChain stakeChain;
 
+        /// <summary>Thread safe access to the best chain of block headers (that the node is aware of) from genesis.</summary>
         private readonly ConcurrentChain chain;
 
+        /// <summary>Consensus' view of UTXO set.</summary>
         private readonly CoinView coinView;
 
+        /// <summary>Defines a set of options that are used by the consensus rules of Proof Of Stake (POS).</summary>
         private readonly PosConsensusOptions consensusOptions;
 
-        static StakeValidator()
-        {
-            ILoggerFactory loggerFactory = new LoggerFactory();
-            loggerFactory.AddNLog();
-            clogger = loggerFactory.CreateLogger(typeof(StakeValidator).FullName);
-        }
-
+        /// <inheritdoc />
+        /// <param name="network">Specification of the network the node runs on - regtest/testnet/mainnet.</param>
+        /// <param name="stakeChain">Database of stake related data for the current blockchain.</param>
+        /// <param name="chain">Chain of headers.</param>
+        /// <param name="coinView">Used for getting UTXOs.</param>
+        /// <param name="loggerFactory">Factory for creating loggers.</param>
         public StakeValidator(Network network, StakeChain stakeChain, ConcurrentChain chain, CoinView coinView, ILoggerFactory loggerFactory)
         {
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
-            this.network = network;
             this.stakeChain = stakeChain;
             this.chain = chain;
             this.coinView = coinView;
             this.consensusOptions = network.Consensus.Option<PosConsensusOptions>();
         }
 
-        public void CheckProofOfStake(ContextInformation context, ChainedBlock pindexPrev, BlockStake prevBlockStake, Transaction tx, uint headerBits)
+        /// <summary>
+        /// Gets the last block in the chain that was generated using
+        /// PoS if <see cref="proofOfStake" /> is <c>true</c> or PoW if <see cref="proofOfStake" /> is <c>false</c>.
+        /// </summary>
+        /// <param name="stakeChain">Chain that is used for retrieving blocks.</param>
+        /// <param name="index">Block that we start from. Only blocks before that one will be checked.</param>
```

Comment:
as mentioned elsewhere, this "Index" naming should probably be changed,
maybe we can do it as a part of this PR as well, the whole method name should be changed
this could be something like `startChainedBlock` and the method name could be GetLastPowPosChainedBlock